bedevere/awaiting.py

```diff
@@ -0,0 +1,219 @@
+  "Awaiting core review" -> "Awaiting merge" [label="New review approves", color=green]
+
+  "New PR" -> "Awaiting merge" [label="New PR", color=green]
+}
+"""
+
+import datetime
+import enum
+import operator
+import random
+
+import gidgethub.routing
+
+
+router = gidgethub.routing.Router()
+
+REQUEST_CHANGE_REVIEW = "I didn't expect the Spanish Inquisition"
+
+TAG_NAME = "changes-requested"
+
+CHANGES_REQUESTED_MESSAGE = f"""\
+<!-- {TAG_NAME}: {{core_dev}} -->
+A Python core developer, {{core_dev}}, has requested some changes be
+made to your pull request before we can consider merging it. If you
+could please address their requests along with any other requests in
+other reviews from core developers that would be appreciated.
+
+Once you have made the requested changes, please leave a comment
+on this pull request containing the phrase `{REQUEST_CHANGE_REVIEW}!`
+I will then notify {{core_dev}} along with any other core developers
+who have left a review that you're ready for them to take another look
+at this pull request.
+<!-- /{TAG_NAME} -->
+
+{{easter_egg}}
+"""
+
+EASTER_EGG_1 = """\
+And if you don't make the requested changes,
+[you will be poked with soft cushions!](https://www.youtube.com/watch?v=Nf_Y4MbUCLY&feature=youtu.be&t=4m7s)
+"""
+
+EASTER_EGG_2 = """\
+And if you don't make the requested changes,
+[you will be put in the comfy chair!](https://www.youtube.com/watch?v=Nf_Y4MbUCLY&feature=youtu.be&t=4m7s)
+"""
+
+CHANGE_REVIEW_REQUESTED = """\
+[Nobody expects the Spanish Inquisition!](https://youtu.be/Nf_Y4MbUCLY)
+
+{core_devs}: please review the changes made to this pull request.
+"""
+
+
+LABEL_PREFIX = "awaiting"
+
+@enum.unique
+class Blocker(enum.Enum):
+    """What is blocking a pull request from being committed."""
+    review = f"{LABEL_PREFIX} review"
+    core_review = f"{LABEL_PREFIX} core review"
+    changes = f"{LABEL_PREFIX} changes"
+    change_review = f"{LABEL_PREFIX} change review"
+    merge = f"{LABEL_PREFIX} merge"
+
+
+def user_login(item):
+    return item["user"]["login"]
+
+
+async def stage(gh, item, blocked_on):
+    """Remove any "awaiting" labels and apply the specified one."""
+    if "issue_url" in item:
+        issue = await gh.getitem(item["issue_url"])
+    else:
+        issue = item
+    label_name = blocked_on.value
+    if any(label_name == label["name"] for label in issue["labels"]):
+        return
+    # There's no reason to expect there to be multiple "awaiting" labels on a
+    # single pull request, but just in case there are we might as well clean
+    # up the situation when we come across it.
+    for label in issue["labels"]:
+        stale_name = label["name"]
+        if stale_name.startswith(LABEL_PREFIX + " "):
+            await gh.delete(issue["labels_url"], {"name": stale_name})
+    await gh.post(issue["labels_url"], data=[label_name])
+
+
+async def is_core_dev(gh, username):
```

Comment:
There's also the option of using https://developer.github.com/v3/orgs/teams/#list-team-members which should also minimize the amount of calls necessary (although I still bet the caching will do the trick since Bedevere is set to cache over 200 MB of cached data).